src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.cpp

```diff
@@ -101,28 +102,45 @@ CHIP_ERROR WiFiDiagnosticsServerCluster::Attributes(const ConcreteClusterPath &
 {
     AttributeListBuilder attributeListBuilder(builder);
 
-    const DataModel::AttributeEntry mandatoryAttributes[] = {
-        Bssid::kMetadataEntry,         SecurityType::kMetadataEntry, WiFiVersion::kMetadataEntry,
-        ChannelNumber::kMetadataEntry, Rssi::kMetadataEntry,
-    };
-
     const BitFlags<WiFiNetworkDiagnostics::Feature> featureFlags = mLogic.GetFeatureFlags();
 
-    const bool hasErrorCounts  = featureFlags.Has(Feature::kErrorCounts);
-    const bool hasPacketCounts = featureFlags.Has(Feature::kPacketCounts);
-    // Define optional attributes based on features
-    const AttributeListBuilder::OptionalAttributeEntry optionalEntries[] = {
-        { hasErrorCounts, BeaconLostCount::kMetadataEntry },
-        { hasErrorCounts, OverrunCount::kMetadataEntry },
-        { hasPacketCounts, BeaconRxCount::kMetadataEntry },
-        { hasPacketCounts, PacketMulticastRxCount::kMetadataEntry },
-        { hasPacketCounts, PacketMulticastTxCount::kMetadataEntry },
-        { hasPacketCounts, PacketUnicastRxCount::kMetadataEntry },
-        { hasPacketCounts, PacketUnicastTxCount::kMetadataEntry },
-        { mLogic.GetEnabledAttributes().enableCurrentMaxRate, CurrentMaxRate::kMetadataEntry },
+    const DataModel::AttributeEntry optionalAttributes[] = {
+        BeaconLostCount::kMetadataEntry,        //
+        OverrunCount::kMetadataEntry,           //
+        BeaconRxCount::kMetadataEntry,          //
+        PacketMulticastRxCount::kMetadataEntry, //
+        PacketMulticastTxCount::kMetadataEntry, //
+        PacketUnicastRxCount::kMetadataEntry,   //
+        PacketUnicastTxCount::kMetadataEntry,   //
+        CurrentMaxRate::kMetadataEntry,         //
     };
 
-    return attributeListBuilder.Append(Span(mandatoryAttributes), Span(optionalEntries));
+    chip::app::SupportedAttributes<WiFiNetworkDiagnostics::Attributes::CurrentMaxRate::Id,         //
```

Comment:
Followup: it's a little annoying to have to list the attributes twice: once in the `optionalAttributes` array and once in the template here.

If the optionalAttributes array were constexpr, it feels like we could probably do something where we template on the array and the template machinery extracts the attribute ids from the metadata entries, right?

Again, followup.

Also, in this specific case, is this type here different from WiFiDiagnosticsServerLogic::SupportedAttributes?